MovieJournalMVC/Controller/MasterViewController.swift

```diff
@@ -9,73 +9,82 @@
 import UIKit
 
 class MasterViewController: UITableViewController {
-
+    
     var detailViewController: DetailViewController? = nil
-    var objects = [Any]()
-
-
+    var objects = Store.shared.movieReviews
+    var count = 0 //TODO: delete this variable and its later usage post-testing
+    
     override func viewDidLoad() {
         super.viewDidLoad()
         // Do any additional setup after loading the view, typically from a nib.
         navigationItem.leftBarButtonItem = editButtonItem
-
+        
         let addButton = UIBarButtonItem(barButtonSystemItem: .add, target: self, action: #selector(insertNewObject(_:)))
         navigationItem.rightBarButtonItem = addButton
         if let split = splitViewController {
             let controllers = split.viewControllers
             detailViewController = (controllers[controllers.count-1] as! UINavigationController).topViewController as? DetailViewController
         }
     }
-
+    
     override func viewWillAppear(_ animated: Bool) {
         clearsSelectionOnViewWillAppear = splitViewController!.isCollapsed
         super.viewWillAppear(animated)
     }
-
+    
     @objc
     func insertNewObject(_ sender: Any) {
-        objects.insert(NSDate(), at: 0)
+        
+        let newReview = MovieReview(title: "Movie #\(count)", review: "Desctiption for Movie #\(count)")
+        
+        count += 1
+        
+        objects.insert(newReview, at: 0)
         let indexPath = IndexPath(row: 0, section: 0)
         tableView.insertRows(at: [indexPath], with: .automatic)
     }
-
+    
     // MARK: - Segues
-
+    
     override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
         if segue.identifier == "showDetail" {
```

Comment:
One of the bad bits of this template: this should really be a guard, until you have multiple segues to handle.

Reply:
I created an enum to switch on segue identifiers...currently there is only one enum case, but there are the benefits that come with a strong data type.